### DISCUSSION TITLE
RFC on Rewards and Recognition module [New Feature ideation]

### DISCUSSION THREAD

**Discussion by luckynvdu** [Sunbird-Ed/Community#29] [General Discussion]
The following is an ideation (initial) for enhancing the rewards and recognition feature. Requesting the community to go through the same and share their thoughts/stimulating ideas/challenges/risks.

**Summary**
A reward is generally a gift given to a user from the platform to celebrate the accomplishment of the user in a platform. Rewards when tied to user activities, can encourage users of the platform to do more. Rewards tend to target specific needs of the platform usage:
a) User Longevity/retention
b) User Engagement
c) User Activities

Recognition is more of a platform acknowledging user identity with accomplishments.
a) Highest Ranked Users
b) User Rankings based on Plays, activities, shares, etc.
c) Assigning Users with personas/badges based on experience in the platform. 

**Problem Statement/Need Statement**

SunbirdED or its adopters today lack a mechanism to continuously engage users along with loyalty. The platform can be extended with a mechanism to leverage the user enrollment data and other useful information to reward the users.

SunbirdED can don't have any other channels of rewarding users other than certificates which can motivate the users to use the platform in a better way.

**Solution**

SunbirdED can be extended to associate rewards against the profile of the user using his enrollment data and other useful information that might exist on the platform.

Ex: Users can be rewarded for first-course consumption on the platform. Similarly, any number of Configurable rewards can be set at the platform level in order to engage users.

SunbirdED can facilitate users to be provided with rankings that can be retained based on the number of plays.

**Future Ideation**

SunbirdED can leverage the rewards and recognition program to enable real-time rewards for the users.
a) Discount Coupons of Popular Brands
b) Cashback Rewards
c) Shopping Coupons etc.

@pramodkvarma @madhucr @alok-os @gauravgupta @kameshbhr @mmanimegalai @SanthoshVasabhaktula @vrayulu @pritamps @karn-nahata @pkranga @Prerna789 @aayushkedia95 @vijiurs @aishwaryashikshalokam @Prateek-slokam @aks30 @varsha-sl @MeghnaSL @rhwarrier @anandp504 @reshmi-nair @Krishnaj20 @pallakartheekreddy @vinukumar-vs @sharathkashyap

**Comment by surendrasinghs** [comment]
I think this is an essential capability for any engaging learning solution/platform/product. It amplifies the innate human need to feel recognised/appreciated and the need to share/celebrate.

As we begin to think of this we should keep in mind the following aspects
1. Wide variety of rewards and recognitions that can be provided beyond just certificates. I mean, the representation could be of various forms - badges, stickers, ribbons, hat/cap, coins/stars, or any object/thing.
2. Configurable rules for issuing these rewards and recognition based on the strategic need/goal and to nudge specific user behaviours. For example, profile set up completed, first QR scan, first content completed, first course completed, first time rating provided, etc.
3. Ability to access/view and share my achievements + position them prominently to keep reminding me of them. For example, my learning passbook, my trophy case, etc
4. Downstream benefits of these rewards and recognitions so that as a user I want to keep collecting them. For example, position in the leaderboard, redeem coins to get 5 new content/course recommendations, unlock a new theme, unlock personal dashboard, etc. These need not necessarily be monetary benefits or any partner service provider benefits that assume/require/work only in a specific context but rather something generalised that leads to more interactions and engagement on the app/platform.
5. Clear and transparent rules on what actions result in some reward/recognition and a list of all possible rewards I can earn.

That said... We should start with something simple - issue a profile completed badge, issue a badge on completion of activities in a group, or issue a completion certificate for every course (even when course mentor didn't enable the certificate), etc.

**Reply by Krishnaj20** [reply]
Adding to @surendrasinghs 's suggestions:
6.  The interoperability & verifiability of these rewards and recognition is also important - should we leverage Sunbird RC spec?

**Reply by Krishnaj20** [reply]
@luckynvdu - can you expand more on the need/problem statement by calling out specific growth loops you envision and how they impact the retention or engagement metrics. For example., one loop could be to increase the course completion rate of users. In this loop: 
a) user completes course module
b) system rewards the user for completing the course module 
c) user is motivated by the reward and is more likely to complete the next module   
Thereby, increasing the course completion rate. 
If you can include more such examples illustrating how 'rewards & recognition' drive retention of SunbirdEd, it can amplify the need and can also provide in what context are you envisioning this need.

**Comment by pramodkvarma** [comment]
User wallet (to account points, usage, etc) should be a separate building block. User profile can be integrated to it and based on the wallet coupons rules (like our UPI mandate), spending it can be controlled (how/where/when).

Recognition is completely separate and Sunbird-ED already has integrated Sunbird-RC that supports configuring, issuing, storing, verifying various types of credentials used for recognition purposes. This is already integrated with Sunbird-ED user profile as I understand.

@luckynvdu @Krishnaj20 @surendrasinghs

**Comment by aishwaryashikshalokam** [comment]
Hello Everyone!

I hope everyone is doing great!

We have been brainstorming on what the platform can offer in terms of rewards and recognition to the users so that they are more motivated and excited to take up resources on the platform and engage more with the platform.
We have put down a high-level solution note on the same.
PFA link of the document:  https://docs.google.com/document/d/1KaP-1GkKDCXEsMPbQld8HYi8PoqiyqnO2_SMhMtxT2A/edit?usp=sharing

The main problem statement that we identified with programs, was the low completion rate of resources on the platform. 
And our solution is focused on the incentives that the platform can provide that enable users to complete them and consume more.

We also tested the solution with 75 users and saw an increase in the completion rate.

It will be great if all of us can share our inputs and suggestions on the proposed solution. Feel free to leave your comments in the solution note itself.
Based on everyone's inputs, we can take it up in development.

@Prateek-slokam @aks30 @kiranharidas187 @vijiurs @Sanjana-dm @varsha-sl @luckynvdu @surendrasinghs @rhwarrier 
@pramodkvarma @madhucr @alok-os @GauravGupta @kameshbhr @mmanimegalai @SanthoshVasabhaktula @vrayulu @pritamps @karn-nahata @pkranga @Prerna789 @aayushkedia95 @anandp504 @reshmi-nair @pallakartheekreddy @vinukumar-vs @sharathkashyap

**Comment by aishwaryashikshalokam** [comment]
Hello Everyone!

This a gentle reminder to add your suggestions, feedback, and questions in the above attached high-level solution note on what the platform can offer in terms of rewards and recognition.

**Reply by Krishnaj20** [reply]
@aishwaryashikshalokam - thanks for initiating this thread.

In my view, we are talking about a new gamification/incentivization service which will enable us to take the elements from games and integrate them to create rich and engaging experiences for the end-users. It is essentially harnessing all the excitement and perseverance people exhibit while playing a game into non-game situations.

The major objective of this service is:
a) To fuel organic growth
b) To incentivise good platform behaviour leading to more stickiness and engagement

This is a great capability which can be utilized for a variety of use cases. In order to make it truly applicable for a variety of use cases, I suggest you abstract it in the solution doc itself and then go specific into the use cases relevant to you. I have added my inputs on how you can generalize this specific ask at the end of [solution](https://docs.google.com/document/d/1KaP-1GkKDCXEsMPbQld8HYi8PoqiyqnO2_SMhMtxT2A/edit) doc. Please go through it. 

As part of the RFC process, I believe someone from the working group will assist you to come up with a PRD and design doc - which can then be submitted for community feedback. I'm more than glad to help you come up with the PRD.

@sharathkashyap - do let @aishwaryashikshalokam know if you/someone from the working group can help her/team to come up with the design doc?

@madhucr @alok-os @SanthoshVasabhaktula @vrayulu

**Comment by xiaoliangliu** [comment]
您好，您的邮件我已收到，我会尽快查收并回复，谢谢
